### PR TITLE
Aml 56 minority sensitivity

### PR DESCRIPTION
**Pull request by nielsuit227**

### CODE REVIEW

**Review by tstreule — CHANGES_REQUESTED** [review]
Minor changes requested.

File: Amplo/AutoML/Documenter.py [hunk 1]
```diff
@@ -246,9 +267,12 @@ def binary_markdown(self):
...
 | F1 Score    | {:.2f} \u00B1 {:.2f} % |""".format(np.mean(accuracy), np.std(accuracy),
-                                                   np.mean(precision), np.std(accuracy),
-                                                   np.mean(sensitivity), np.std(sensitivity),
-                                                   np.mean(specificity), np.std(specificity),
+                                                   np.mean(precision), np.std(
+                                                       accuracy),
+                                                   np.mean(sensitivity), np.std(
+                                                       sensitivity),
+                                                   np.mean(specificity), np.std(
+                                                       specificity),
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/AutoML/Documenter.py:275]: 106819f

File: Amplo/AutoML/FeatureProcessor.py [hunk 2]
```diff
@@ -446,13 +449,13 @@ def _fit_cross_features(self):
         for i, key_a in enumerate(tqdm(self.non_date_cols)):
             accepted_for_key_a = 0
             for j, key_b in enumerate(random.sample(self.non_date_cols, len(self.non_date_cols))):
-                # Skip if they're the same
+                # Skip if they"re the same
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/AutoML/FeatureProcessor.py:452]: 38da7dc

File: Amplo/AutoML/FeatureProcessor.py [hunk 3]
```diff
@@ -528,33 +531,33 @@ def _add_trigonometry_features(self):
...
-            print('[AutoML] Analysing Linear Features'.format(len(self.linearFeatures)))
+            logger.info("Analysing Linear Features".format(len(self.linearFeatures)))
 
         scores = {}
         start_time = time.time()
         for i, key_a in enumerate(self.non_date_cols):
             accepted_for_key_a = 0
             for j, key_b in enumerate(random.sample(self.non_date_cols, len(self.non_date_cols))):
-                # Skip if they're the same
+                # Skip if they"re the same
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/AutoML/FeatureProcessor.py:554]: 38da7dc

File: Amplo/AutoML/FeatureProcessor.py [hunk 4]
```diff
@@ -528,33 +531,33 @@ def _add_trigonometry_features(self):
...
                 if key_a == key_b:
                     continue
                 # Skip rest if key_a is not useful in first max(50, 30%) (uniform)
                 if accepted_for_key_a == 0 and j > max(50, int(len(self.non_date_cols) / 3)):
                     continue
-                # Skip if we're out of time
+                # Skip if we"re out of time
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/AutoML/FeatureProcessor.py:560]: 38da7dc

File: Amplo/Documenting/RegressionDocumenting.py [hunk 5]
```diff
@@ -69,12 +70,12 @@ def analyse(self):
...
-        print('[AutoML] Mean Squared Error:             {:.2f} \u00B1 {:.2f}'.format(np.mean(mse), np.std(mse)))
-        print('[AutoML] R2 Score:                       {:.2f} \u00B1 {:.2f}'.format(np.mean(r2), np.std(r2)))
-        print('[AutoML] Max Error:                      {:.2f} \u00B1 {:.2f}'.format(
+        logger.info('[AutoML] Mean Absolute Error:            {:.2f} \u00B1 {:.2f}'.format(np.mean(mae), np.std(mae)))
+        logger.info('[AutoML] Mean Squared Error:             {:.2f} \u00B1 {:.2f}'.format(np.mean(mse), np.std(mse)))
+        logger.info('[AutoML] R2 Score:                       {:.2f} \u00B1 {:.2f}'.format(np.mean(r2), np.std(r2)))
+        logger.info('[AutoML] Max Error:                      {:.2f} \u00B1 {:.2f}'.format(
             np.mean(max_error), np.std(max_error)))
-        print('[AutoML] Mean Absolute Percentage Error: {:.2f} \u00B1 {:.2f}'.format(
+        logger.info('[AutoML] Mean Absolute Percentage Error: {:.2f} \u00B1 {:.2f}'.format(
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/Documenting/RegressionDocumenting.py:78]: 311ca40

File: Amplo/AutoML/IntervalAnalyser.py [hunk 6]
```diff
@@ -108,7 +108,7 @@ def data_with_noise(self) -> pd.DataFrame:
 
         Notes
         -----
-        Depending on the dtype of the noise attribute, the dtype of the labels' column will be affected.
+        Depending on the dtype of the noise attribute, the dtype of the labels" column will be affected.
```
[the remark this answers is not in the document]
Comment (nielsuit227) [reply Amplo/AutoML/IntervalAnalyser.py:111]: 311ca40

File: Amplo/Pipeline.py [hunk 7]
```diff
@@ -51,113 +51,113 @@ def __init__(self, **kwargs):
...
+        process_data : bool, optional
             Whether to force data processing
-        document_results : bool
+        document_results : bool, optional
             Whether to force documenting
-        no_dirs : bool
+        no_dirs : bool, optional
             Whether to create files or not
-        verbose : int
+        verbose : int, optional
```
Comment (tstreule) [inline Amplo/Pipeline.py:160]: Type-hinting `optional` for all these parameters is malicious. I know what you mean with "optional" (they have a default value) but by saying it's optional, you encourage users to set parameters to `None` which we do not accept.
Therefore, remove all optionals, except for `logging_level` which actually can handle `None` as input.

**Review by tstreule — CHANGES_REQUESTED** [review]

File: Amplo/Pipeline.py (hunk 7 above)
Comment (nielsuit227) [reply Amplo/Pipeline.py]: https://github.com/Amplo-GmbH/AutoML/pull/32/commits/2fa3f943f73e27c00b0d0269666db0f858de4c68

**Comment by nielsuit227** [comment]
Passing again, ready for review :)

**Review by tstreule — APPROVED** [review]

**Pull request merged**